### PR TITLE
CompanyService_DisablePositionById

### PR DESCRIPTION
**Pull request by lalagvana**
Added DisablePositionById endpoint and its tests.

### CODE REVIEW

**Review by KostinP — APPROVED** [review]
Add migration.

File: Services/CompanyService/Commands/Interfaces/IDisablePositionByIdCommand.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    /// Provides method for deleting a position.
+    /// </summary>
+    public interface IDisablePositionByIdCommand
+    {
+        /// <summary>
+        ///  Deletes an existing position. Returns true if it succeeded to delete a position, otherwise Exception.
+        /// </summary>
+        /// <param name="positionId">Position Id.</param>
+        /// <returns>New position Id.</returns>
+        bool Execute(Guid positionId);
```
Comment (DarkFelas) [inline Services/CompanyService/Commands/Interfaces/IDisablePositionByIdCommand.cs]: no need for bool. just **void**

File: Services/CompanyService/Repositories/PositionRepository.cs [hunk 2]
```diff
@@ -47,6 +47,20 @@ public DbPosition GetUserPosition(Guid userId)
...
+            var dbPosition = dbContext.Positions.FirstOrDefault(position => position.Id == positionId);
+            if (dbPosition == null)
+            {
+                throw new Exception("Position with this id was not found.");
+            }
+            dbPosition.IsActive = false;
+            dbContext.Positions.Update(dbPosition);
+            dbContext.SaveChanges();
+
+            return true;
```
Comment (DarkFelas) [inline Services/CompanyService/Repositories/PositionRepository.cs]: Remove, no need for bool

**Review by DarkFelas — APPROVED** [review]

File: Services/CompanyService/Repositories/PositionRepository.cs [hunk 3]
```diff
@@ -47,6 +47,18 @@ public DbPosition GetUserPosition(Guid userId)
             return dbContext.Positions.Find(dbCompanyUser.PositionId);
         }
 
+        public void DisablePositionById(Guid positionId)
+        {
+            var dbPosition = dbContext.Positions.FirstOrDefault(position => position.Id == positionId);
+            if (dbPosition == null)
```
Comment (djino97) [inline Services/CompanyService/Repositories/PositionRepository.cs]: add emty line

File: Services/CompanyService/Repositories/PositionRepository.cs [hunk 4]
```diff
@@ -47,6 +47,18 @@ public DbPosition GetUserPosition(Guid userId)
...
+            {
+                throw new Exception("Position with this id was not found.");
+            }
+            dbPosition.IsActive = false;
```
Comment (djino97) [inline Services/CompanyService/Repositories/PositionRepository.cs]: add emty line

**Review by djino97 — APPROVED** [review]

File: Services/CompanyService/Commands/DisablePositionByIdCommand.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+        public DisablePositionByIdCommand([FromServices] IPositionRepository repository)
+        {
+            this.repository = repository;
+        }
+
+        public void Execute(Guid positionId)
+        {
+            repository.DisablePositionById(positionId);
+        }
```
Comment (denis-kriv) [inline Services/CompanyService/Commands/DisablePositionByIdCommand.cs:20]: what if positionId will be null or empty?

Comment (lalagvana) [reply Services/CompanyService/Commands/DisablePositionByIdCommand.cs:20]: There'll be exception in Repository. I've added tests for it.

Comment (mekk1t) [reply Services/CompanyService/Commands/DisablePositionByIdCommand.cs:20]: Maybe we should add checks? Like does the user have rights or is he an admin. Use AccessValidator from develop, but don't forget to register it and modify the appsettings.json properly.

File: Services/CompanyService/Database/Migrations/20200823110601_NewDbFileModel.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace LT.DigitalOffice.CompanyService.Database.Migrations
+{
+    public partial class NewDbFileModel : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
```
Comment (mekk1t) [inline Services/CompanyService/Database/Migrations/20200823110601_NewDbFileModel.cs]: The name of the migration should be NewDbPosition model, isn't it?

**Review by mekk1t — APPROVED** [review]
Check out the comments

File: Services/CompanyService/Database/Migrations/20200823110601_NewDbFileModel.cs (hunk 6 above)
Comment (lalagvana) [reply Services/CompanyService/Database/Migrations/20200823110601_NewDbFileModel.cs]: fixed

**Pull request merged**